build/generate/docuTOC.go

```diff
@@ -0,0 +1,156 @@
+	if folderPath == "" {
+		return fmt.Errorf("Empty docutoc path")
+	}
+
+	// Find all the markdown files in the specified folder and its subfolders.
+	markdownFiles, err := findMarkdownFiles(folderPath)
+	if err != nil {
+		return err
+	}
+
+	// Sort the list of markdown files in alphabetical order
+	sort.Strings(markdownFiles)
+
+	// Create the table of contents.
+	toc := generateTableOfContents(folderPath, markdownFiles)
+
+	err = updateTableOfContents(filepath.Join(folderPath, "SUMMARY.md"), startMarker, endMarker, toc)
+	if err != nil {
+		// log.Fatal(err)
+		return err
+	}
+
+	fmt.Println("Table of contents generated successfully!")
+	return nil
+}
+
+// func stringInSlice(a string, list []string) bool {
+//     for _, b := range list {
+//         if b == a {
+//             return true
+//         }
+//     }
+//     return false
+// }
+
+func generateTableOfContents(folderPath string, markdownFiles []string) string {
+	var toc strings.Builder
+	currentFolder := ""
+
+	// folderExceptions := []string{
+	// 	".DS_Store",
+	// }
+	fileExceptions := []string{
+		"js.md",
+		"index.md",
+		"summary.md",
+	}
+
+	//First sort by folders, then by filename.
+	sort.SliceStable(markdownFiles, func(i, j int) bool {
+		if filepath.Dir(markdownFiles[i]) == filepath.Dir(markdownFiles[j]) {
+			return markdownFiles[i] < markdownFiles[j]
+		} else {
+			return filepath.Dir(markdownFiles[i]) < filepath.Dir(markdownFiles[j])
+		}
+	})
+
+	for _, file := range markdownFiles {
+		//count is folder depth for toc indentation, minus one for docu folder
+		count := strings.Count(file, string(filepath.Separator)) - 1
+		filename := filepath.Base(file)
+		//if the file is an exception listed above, skip it.
+		if stringInSlice(strings.ToLower(filename), fileExceptions) {
+			continue
+		}
+
+		fileFolder := filepath.Dir(file)
+		if fileFolder != currentFolder {
+			//if we are in the documentation root, skip over these files.
+			if folderPath == fileFolder {
+				// continue
+			}
+			currentFolder = fileFolder
+			folderName := filepath.Base(currentFolder)
+			// Captalize folder names, replace underscores with spaces for # headings
+			toc.WriteString(strings.Repeat("    ", count))
+
+			toc.WriteString("* " + strings.Title(strings.ReplaceAll(folderName, "_", " ")) + "\n")
+		}
+		fmt.Println(file)
+
+		// naming exceptions
+		displayfilename := strings.TrimSuffix(filename, filepath.Ext(filename))
+		if !strings.Contains(file, "functions") {
+			displayfilename = strings.Title(strings.ReplaceAll(displayfilename, "_", " "))
+		}
+
+		toc.WriteString(strings.Repeat("    ", count+1))
+		toc.WriteString("* [" + displayfilename + "](" + filepath.Join(".", strings.ReplaceAll(file, folderPath, "")) + ")\n")
+	}
+	return toc.String()
+}
+
+// updateTableOfContents inserts the generated table of contents between the two markers in the specified markdown file.
+func updateTableOfContents(markdownFile, startMarker, endMarker, toc string) error {
```

Comment:
This seems generic enough that it can be re-used.  How about calling it `replaceTextBetweenMarkers()` and change `toc` to `replacement`?

Reply:
And maybe merge/eextract the existing [`replaceInlineContent()`](https://github.com/StackExchange/dnscontrol/blob/v3.28.0/build/generate/featureMatrix.go#L294) marker replacer?

Reply:
Done.

Reply:
What is the reason you left the old implementation uncommented in [21dd26d](https://github.com/StackExchange/dnscontrol/commit/21dd26d7cae1ffecea62d30166c8d757b9ab856f#diff-28e66bede643a7156cf9c19950731e1a3a9ab66becf1a063f7705076bfc4dbdaR293)? As far as I'm concerned, this can be completely removed.